### ISSUE TITLE
Generated TableImpl::getSchema override should be annotated Nullable

### ISSUE THREAD

**Issue by cies** [jOOQ/jOOQ#14431]
### Expected behavior

When adding annotations with:

```kotlin
            withNullableAnnotation(true)
            withNullableAnnotationType("org.jetbrains.annotations.Nullable")
            withNonnullAnnotation(true)
            withNonnullAnnotationType("org.jetbrains.annotations.NotNull")
```

I should still be able to use jOOQ.

### Actual behavior

But it fails with: `TaskTbl.getSchema must not return null`

Looking at the code the bug seems obvious:

```java
    @Override
    @NotNull
    public Schema getSchema() {
        return aliased() ? null : DefaultSchema.DEFAULT_SCHEMA;
    }
```

The `@NotNull` annotation on `getSchema()` should be replaced with `@Nullable` given `return null` is possible.

### Steps to reproduce the problem

Just add these:

```kotlin
            withNullableAnnotation(true)
            withNullableAnnotationType("org.jetbrains.annotations.Nullable")
            withNonnullAnnotation(true)
            withNonnullAnnotationType("org.jetbrains.annotations.NotNull")
```

to any jOOQ project and it will fail on the first query.

### jOOQ Version

3.17.4

### Database product and version

MySQL8

### Java Version

17

### OS Version

Linux

### JDBC driver name and version (include name if unofficial driver)

mysql-connector-j v8.0.30

**Comment by cies** [comment]
I have tried to manually change the `@NotNull` annotation to `@Nullable`, and it fixes the problem.

**Comment by cies** [comment]
Fixed in PR.

**Comment by lukaseder** [comment]
Thanks a lot for your report and PR.

**Comment by lukaseder** [comment]
This is a regression introduced with https://github.com/jOOQ/jOOQ/issues/11115 in jOOQ 3.15.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.7 (https://github.com/jOOQ/jOOQ/issues/14434), and 3.16.13 (https://github.com/jOOQ/jOOQ/issues/14435)
